uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenPaymentPresenter.kt

```diff
@@ -74,48 +96,82 @@ class AdyenPaymentPresenter(view: PaymentMVP.View,
         } ?: view?.showError(R.string.payment_issue_message)
     }
 
-    override fun passBackNonce(sdkNonce: String) {
-        this.sdkToken = sdkNonce
+    override fun handleActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (resultCode == AppCompatActivity.RESULT_OK && data != null) {
+            val dataString = data.getStringExtra(AdyenResultActivity.RESULT_KEY)
+            val payload = JSONObject(dataString)
+            when (payload.optString(AdyenPaymentView.RESULT_CODE, "")) {
+                AdyenPaymentView.AUTHORISED -> {
+                    val transactionId = payload.optString(AdyenPaymentView.MERCHANT_REFERENCE, "")
+                    updateCardDetails(nonce = transactionId,
+                                      paymentData = payload.optString
+                                      (AdyenPaymentView.ADDITIONAL_DATA, null))
+                    passBackNonce(transactionId)
+                }
+                else -> view?.showPaymentFailureDialog()
```

Comment:
I am wondering should we map all other result codes to failure? e.g. Received means it is PENDING. 
It should be a question for settlement and payments team to define what should be the behaviour of FE based on the expectation of BE
https://docs.adyen.com/development-resources/test-cards/result-code-testing/adyen-response-codes

Reply:
So basically just pass back nonce for everything?